lib/hamster/hash.rb

```diff
@@ -590,6 +590,27 @@ def fetch_values(*wanted)
       Vector.new(array.freeze)
     end
 
+    # Return the value of successively indexing into a hash
+    # If any of the keys is not present in the hash, return `hil`
+    #
+    # @example
+    #   h = Hamster::Hash[:a => 9, :b => Hamster::Hash[:c => 'a', :d => 4], :e => nil]
+    #   h.dig(:b, :c) # => "a"
+    #   h.dig(:b, :f) # => nil
+    # @params keys to fetch from the hash
+    # @return value
+    def dig(key, *rest)
+      value = self[key]
+      if rest.empty? || value.nil?
+        value
+      else
+        if value.respond_to?(:dig)
+          value.dig(*rest)
+        end
+      end
+    end
```

Comment:
I don't understand. `Hamster::Vector` *does* raise an exception when `#[]` is called with a non-integer:

```
2.2.3 :002 > v = Hamster::Vector[1,2,3]
 => Hamster::Vector[1, 2, 3] 
2.2.3 :003 > v[1]
 => 2 
2.2.3 :004 > v[:first]
ArgumentError: comparison of Symbol with 0 failed
	from /home/dmurik/world/hamster/lib/hamster/vector.rb:215:in `<'
	from /home/dmurik/world/hamster/lib/hamster/vector.rb:215:in `get'
	from /home/dmurik/world/hamster/lib/hamster/vector.rb:323:in `slice'
	from (irb):4
	from /home/dmurik/.rvm/rubies/ruby-2.2.3/bin/irb:11:in `<main>'
```

So if you use your current suggested `#dig` implementation for Vectors too you'll get a behaviour similar to 2.3.0-preview1. (unless I missed something...)